src/test/ts/part3/Part3Ex2Test.ts

```diff
@@ -1,65 +1,114 @@
+import { TinyAssertions, TinyDom, TinyHooks, TinySelections } from '@ephox/mcagar';
+import { Traverse } from '@ephox/sugar';
+import { assert } from 'chai';
 import { Editor } from 'tinymce';
 
-describe('Part3Ex2Test', () => {
-  /*
-  So you might have noticed that the code we had to write in exercise 1 was a
-  lot of effort. It also could be made nicer in a few ways:
-
-  - If the test fails, leave the editor on-screen so you can look at what went
-    wrong
-  - someone please help me think of cool things about TinyHooks before I merge
-  this
-
-  Enter the TinyHooks module, from mcagar (mc - Tiny[MC]E, agar - another
-  in-house testing library, more on that later). It lets you write the following.
-   */
-  const hook = TinyHooks.bddSetup<Editor>({
-    // And then you put your settings in here
-    toolbar: 'bold',
-    // No need to add a "setup" or "init_instance_callback" here unless you
-    // need one for the actual test you're writing.
+describe('Part3Ex3Test', () => {
+  const hook = TinyHooks.bddSetup<Editor>({ height: '100vh' });
+
+  beforeEach(() => {
+    // Fun fact, hook.editor() also works inside of before / beforeEach
+    // callbacks, as long as you declare the hook first.
+
+    const editor = hook.editor();
+    editor.setContent([
+      '<p>Here is a bit of content</p>',
+      '<p>And some <strong>bolded</strong> content</p>',
+      '<p><img src="https://http.cat/301.jpg" alt="A cat picture" /></p>'
+    ].join(''));
   });
 
-  it('looks like an editor', () => {
+  it('selects content like an editor', () => {
+    const editor = hook.editor();
+
     /*
-    You can use this hook object only inside a test (so that you know the before
-    callback has been run), and you use it like this.
+    Now we're going to talk about one of the more complicated parts of editor
+    development: selections.
+
+    It's very rare you'll need to deal with moving the selection around early
+    in your training, and make sure to chat with a senior dev about what you're
+    doing if you do need to. However, for a lot of tests its important to change
+    the selection, or make assertions based on it.
+
+    Under most circumstances, you can think of the selection as two
+    node/offset pairs. The node tells you which DOM node the selection starts (or ends) in
+    and the offset tells you how far into that node the selection starts (or
+    ends). If your node is a HTMLElement, then the offset is going to tell you
+    how many of the node's ChildNodes are before the selection endpoint. If
+    your node is a Text, then the offset is going to tell you how many
+    characters into the text your selection endpoint is.
+
+    This concept is summed up really nicely by the builtin "Range" API exported
+    by the browser. Let's create a Range object, and make sure it works.
      */
-    const editor = hook.editor();
+    // TinyMCE also exposes this document.createRange() through the `dom` API
```

Comment:
```suggestion
    // TinyMCE exposes the built-in JS document.createRange() through the `dom` API
```
maybe? I only understood this because we talked about it the other day...